pkg/chunkenc/pool.go

```diff
@@ -277,6 +247,9 @@ type BufioReaderPool struct {
 // Get returns a bufio.Reader which reads from r. The buffer size is that of the pool.
 func (bufPool *BufioReaderPool) Get(r io.Reader) *bufio.Reader {
 	buf := bufPool.pool.Get().(*bufio.Reader)
+	if buf == nil {
+		return bufio.NewReaderSize(r, 4*1024)
```

Comment:
unrelated optimization, since this will read lines, we can avoid the need to grow unless line are bigger than 4k bytes. This is also reused.